editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/pages/Wsdl2RestWizardBasePage.java

```diff
@@ -0,0 +1,348 @@
+				}
+				return false;
+			}
+		});
+		dialog.setTitle(UIMessages.wsdl2RestWizardSecondPageContainerSelectionDialogTitle);
+		dialog.setMessage(UIMessages.wsdl2RestWizardSecondPageContainerSelectionDialogMessage);
+		dialog.setInput(project);
+		if (dialog.open() == ElementTreeSelectionDialog.OK) {
+			Object[] result = dialog.getResult();
+			if (result.length == 1) {
+				return (((IFolder) result[0]).getFullPath().toPortableString());
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Opens a dialog to allow selection of a Java class in the workbench.
+	 * @param project
+	 * @param shell
+	 * @return
+	 */
+	protected String handleClassBrowse(IProject project, Shell shell) {
+		IJavaSearchScope scope = null;
+		if (project != null) {
+			IJavaProject jproject = JavaCore.create(project);
+			if (jproject == null) {
+				scope = SearchEngine.createWorkspaceScope();
+			} else {
+				scope = SearchEngine.createJavaSearchScope(new IJavaElement[] { jproject });
+			}
+		}
+
+		try {
+			SelectionDialog dialog = JavaUI.createTypeDialog(shell, null, scope,
+					IJavaElementSearchConstants.CONSIDER_CLASSES_AND_INTERFACES, false, ""); //$NON-NLS-1$
+			if (dialog.open() == SelectionDialog.OK) {
+				Object[] result = dialog.getResult();
+				if (result.length > 0 && result[0] instanceof IType) {
+					return ((IType) result[0]).getFullyQualifiedName();
+				}
+			}
+		} catch (JavaModelException e) {
+			Wsdl2RestUIActivator.pluginLog().logError(e);
+		}
+		return null;
+	}
+
+	/**
+	 * Initialize text control if incoming String is not null or empty.
+	 * @param textControl
+	 * @param value
+	 */
+	protected void initIfNotEmpty (Text textControl, String value) {
+		if (!Strings.isEmpty(value)) {
+			textControl.setText(value);
+		}
+	}
+
+	/**
+	 * Utility method to find the target service URL for the WSDL if it is available. 
+	 * @return String
+	 */
+	protected String getLocationFromWSDL() {
+		String locationURL = null;
+		if (!Strings.isEmpty(getOptionsFromWizard().getWsdlURL())) {
+			URL url;
+			InputStream testStream = null;
+			try {
+				url = new URL(getOptionsFromWizard().getWsdlURL());
+				testStream = url.openStream();
+
+				DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+				DocumentBuilder builder = factory.newDocumentBuilder();
+
+				// Load the input XML document, parse it and return an instance of the
+				// Document class.
+				Document document = builder.parse(testStream);
+
+				// now use XPath to find the particular element we need
+				XPathFactory xpf = XPathFactory.newInstance();
+				XPath xpath = xpf.newXPath();
+				Object rawLocation = 
+						xpath.evaluate("/definitions/service/port/address/@location", document, XPathConstants.STRING);
+
+				// if we found it, stash it!
+				if (rawLocation != null) {
+					locationURL = (String) rawLocation;
+				}
+			} catch (IOException | ParserConfigurationException | SAXException | XPathExpressionException e) {
+				// should be valid at this point
```

Comment:
what do you mean by "should be valid at this point"? although it is in a catch exception block?

Reply:
I mean we shouldn't hit this if there's a URL specified in the WSDL. If not, then we have to deal with that case.